### PR TITLE
master tables curd operations

### PR DESCRIPTION
**Pull request by sachin-chamute**

### CODE REVIEW

File: ClariseMain/src/main/java/com/clairvoyant/clarise/configuration/AuthorizationFilter.java [hunk 1]
```diff
@@ -32,31 +32,23 @@ protected void doFilterInternal(HttpServletRequest request, HttpServletResponse
         String authToken = httpRequest.getHeader("Authorization");
         // TODO: sometime token is coming as null
 
-        if (!StringUtils.hasLength(authToken)) {
```
Comment (santoshgaikwad14) [inline ClariseMain/src/main/java/com/clairvoyant/clarise/configuration/AuthorizationFilter.java:35]: Revert this change,
Remove comments, if you added this comment to test it locally, make sure you are not pushing unnecessary classes.

File: ClariseMain/src/main/java/com/clairvoyant/clarise/exceptions/DesignationNotFoundException.java [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+public class DesignationNotFoundException extends RuntimeException{
+
+	private static final long serialVersionUID = 1L;
+	
+	public DesignationNotFoundException(String message) {
+		super(message);
+	}
+
+}
```
Comment (santoshgaikwad14) [inline ClariseMain/src/main/java/com/clairvoyant/clarise/exceptions/DesignationNotFoundException.java:12]: Instead of creating a separate XxxNotFoundException its good to create common Exception names as ResourceNotFoundException with two methods like below : 

- ResourceNotFoundException(String message)
- ResourceNotFoundException(String message, Throwable cause)

We can use the same [here](https://github.com/teamclairvoyant/skills-base/blob/ee38f82e44caa8e181c83540e0338b7f19775370/ClariseMain/src/main/java/com/clairvoyant/clarise/service/DesignationServiceImpl.java#L49).

File: ClariseMain/src/main/java/com/clairvoyant/clarise/service/DesignationService.java [hunk 3]
```diff
@@ -0,0 +1,17 @@
+package com.clairvoyant.clarise.service;
+
+import java.util.List;
+
+import com.clairvoyant.clarise.model.Designation;
+
+
+public interface DesignationService {
```
Comment (santoshgaikwad14) [inline ClariseMain/src/main/java/com/clairvoyant/clarise/service/DesignationService.java:8]: Rename all interfaces with prefix `I` eg:  IDesignationService - So we can understand it is interface by its name.

File: ClariseMain/src/main/java/com/clairvoyant/clarise/service/QualificationStatusService.java [hunk 4]
```diff
@@ -0,0 +1,17 @@
+package com.clairvoyant.clarise.service;
+
+import java.util.List;
+
+import com.clairvoyant.clarise.model.QualificationStatus;
+
+
+public interface QualificationStatusService {
```
Comment (santoshgaikwad14) [inline ClariseMain/src/main/java/com/clairvoyant/clarise/service/QualificationStatusService.java:8]: Rename interfaces add prefix `I` eg:  IQualificationStatusService

File: ClariseMain/src/main/java/com/clairvoyant/clarise/service/SkillsRatingService.java [hunk 5]
```diff
@@ -0,0 +1,17 @@
+package com.clairvoyant.clarise.service;
+
+import java.util.List;
+import java.util.Optional;
+
+import com.clairvoyant.clarise.model.SkillsRating;
+
+public interface SkillsRatingService {
```
Comment (santoshgaikwad14) [inline ClariseMain/src/main/java/com/clairvoyant/clarise/service/SkillsRatingService.java:8]: Rename interfaces add prefix `I` eg:  ISkillsRatingService

File: ClariseMain/src/main/java/com/clairvoyant/clarise/service/DesignationServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,67 @@
+package com.clairvoyant.clarise.service;
```
Comment (santoshgaikwad14) [inline ClariseMain/src/main/java/com/clairvoyant/clarise/service/DesignationServiceImpl.java:1]: Move all service implementation classes inside `com.clairvoyant.clarise.service.impl.`

File: ClariseMain/src/main/java/com/clairvoyant/clarise/service/QualificationStatusServiceImpl.java [hunk 7]
```diff
@@ -0,0 +1,69 @@
+package com.clairvoyant.clarise.service;
```
Comment (santoshgaikwad14) [inline ClariseMain/src/main/java/com/clairvoyant/clarise/service/QualificationStatusServiceImpl.java:1]: Move all service implementation classes inside `com.clairvoyant.clarise.service.impl.`

File: ClariseMain/src/main/java/com/clairvoyant/clarise/service/SkillsRatingServiceImpl.java [hunk 8]
```diff
@@ -0,0 +1,70 @@
+package com.clairvoyant.clarise.service;
```
Comment (santoshgaikwad14) [inline ClariseMain/src/main/java/com/clairvoyant/clarise/service/SkillsRatingServiceImpl.java:1]: Move all service implementation classes inside `com.clairvoyant.clarise.service.impl.`

File: ClariseMain/src/main/java/com/clairvoyant/clarise/controller/DesignationController.java [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+
+@RestController
+@RequestMapping("/v1/designation")
+public class DesignationController {
+
+	@Autowired
+	private DesignationService service;
+
+	@PostMapping
+	public Designation addDesignation(@RequestBody Designation designation) {
```
Comment (santoshgaikwad14) [inline ClariseMain/src/main/java/com/clairvoyant/clarise/controller/DesignationController.java:27]: Swagger API documentation is missing . please add proper api documentation using `@ApiOperation`

File: ClariseMain/src/main/java/com/clairvoyant/clarise/controller/QualificationStatusController.java [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+import com.clairvoyant.clarise.service.QualificationStatusService;
+
+@RestController
+@RequestMapping("/v1/qualificationstatus")
+public class QualificationStatusController {
+
+	@Autowired
+	private QualificationStatusService service;
+
+	@PostMapping
```
Comment (santoshgaikwad14) [inline ClariseMain/src/main/java/com/clairvoyant/clarise/controller/QualificationStatusController.java:26]: Swagger API documentation is missing . please add proper api documentation using `@ApiOperation`

File: ClariseMain/src/main/java/com/clairvoyant/clarise/controller/SkillsRatingController.java [hunk 11]
```diff
@@ -0,0 +1,48 @@
...
+
+@RestController
+@RequestMapping("/v1/skillsrating")
+public class SkillsRatingController {
+
+	@Autowired
+	private SkillsRatingService service;
+
+	@PostMapping()
+	public SkillsRating addRating(@RequestBody SkillsRating skillsRating) {
```
Comment (santoshgaikwad14) [inline ClariseMain/src/main/java/com/clairvoyant/clarise/controller/SkillsRatingController.java:26]: Swagger API documentation is missing . please add proper api documentation using `@ApiOperation`

File: ClariseMain/src/main/java/com/clairvoyant/clarise/controller/SkillsRatingController.java [hunk 12]
```diff
@@ -0,0 +1,48 @@
...
+import com.clairvoyant.clarise.service.SkillsRatingService;
+
+@RestController
+@RequestMapping("/v1/skillsrating")
+public class SkillsRatingController {
+
+	@Autowired
+	private SkillsRatingService service;
+
+	@PostMapping()
```
Comment (santoshgaikwad14) [inline ClariseMain/src/main/java/com/clairvoyant/clarise/controller/SkillsRatingController.java:25]: Does this () really required here ?

File: ClariseMain/src/main/java/com/clairvoyant/clarise/service/DesignationService.java (hunk 3 above)
Comment (ankit11088) [reply ClariseMain/src/main/java/com/clairvoyant/clarise/service/DesignationService.java:8]: I would suggest not to do that. That is a convention followed in C# I believe, not in java.
For distinguishing, we can have the interface as it is and the implementation as `Default+<interface_name>.java`

For other implementations, which are specific, `Specific_Name+<interface_name>.java`

**Comment by ankit11088** [comment]
Get rid of whitespaces please. It is making changes come in unneccessarily.

File: ClariseMain/src/main/java/com/clairvoyant/clarise/advice/ExceptionController.java [hunk 13]
```diff
@@ -0,0 +1,29 @@
...
+import org.springframework.web.bind.annotation.RestControllerAdvice;
+
+import com.clairvoyant.clarise.exceptions.DesignationNotFoundException;
+import com.clairvoyant.clarise.exceptions.QualificationStatusNotFoundException;
+import com.clairvoyant.clarise.exceptions.SkillRatingNotFoundException;
+
+@RestControllerAdvice
+public class ExceptionController {
+
+	@ExceptionHandler(value = SkillRatingNotFoundException.class)
```
Comment (ankit11088) [inline ClariseMain/src/main/java/com/clairvoyant/clarise/advice/ExceptionController.java:15]: I feel this is code duplication for 404 status.
Better if we go with a ResponseStatusExceptionHandler which takes status code and exception body as argument and throws it.

File: ClariseMain/src/main/java/com/clairvoyant/clarise/configuration/AuthorizationFilter.java (hunk 1 above)
Comment (sachin-chamute) [reply ClariseMain/src/main/java/com/clairvoyant/clarise/configuration/AuthorizationFilter.java:35]: resolved

File: ClariseMain/src/main/java/com/clairvoyant/clarise/exceptions/DesignationNotFoundException.java (hunk 2 above)
Comment (sachin-chamute) [reply ClariseMain/src/main/java/com/clairvoyant/clarise/exceptions/DesignationNotFoundException.java]: resolved

File: ClariseMain/src/main/java/com/clairvoyant/clarise/service/DesignationService.java (hunk 3 above)
Comment (sachin-chamute) [reply ClariseMain/src/main/java/com/clairvoyant/clarise/service/DesignationService.java:8]: resolved

File: ClariseMain/src/main/java/com/clairvoyant/clarise/service/QualificationStatusService.java (hunk 4 above)
Comment (sachin-chamute) [reply ClariseMain/src/main/java/com/clairvoyant/clarise/service/QualificationStatusService.java:8]: resolved

File: ClariseMain/src/main/java/com/clairvoyant/clarise/service/SkillsRatingService.java (hunk 5 above)
Comment (sachin-chamute) [reply ClariseMain/src/main/java/com/clairvoyant/clarise/service/SkillsRatingService.java:8]: resolved

File: ClariseMain/src/main/java/com/clairvoyant/clarise/service/DesignationServiceImpl.java (hunk 6 above)
Comment (sachin-chamute) [reply ClariseMain/src/main/java/com/clairvoyant/clarise/service/DesignationServiceImpl.java]: resolved

File: ClariseMain/src/main/java/com/clairvoyant/clarise/service/QualificationStatusServiceImpl.java (hunk 7 above)
Comment (sachin-chamute) [reply ClariseMain/src/main/java/com/clairvoyant/clarise/service/QualificationStatusServiceImpl.java]: resolved

File: ClariseMain/src/main/java/com/clairvoyant/clarise/service/SkillsRatingServiceImpl.java (hunk 8 above)
Comment (sachin-chamute) [reply ClariseMain/src/main/java/com/clairvoyant/clarise/service/SkillsRatingServiceImpl.java]: resolved

File: ClariseMain/src/main/java/com/clairvoyant/clarise/controller/DesignationController.java (hunk 9 above)
Comment (sachin-chamute) [reply ClariseMain/src/main/java/com/clairvoyant/clarise/controller/DesignationController.java:31]: resolved

File: ClariseMain/src/main/java/com/clairvoyant/clarise/controller/QualificationStatusController.java (hunk 10 above)
Comment (sachin-chamute) [reply ClariseMain/src/main/java/com/clairvoyant/clarise/controller/QualificationStatusController.java:29]: resolved

File: ClariseMain/src/main/java/com/clairvoyant/clarise/controller/SkillsRatingController.java (hunk 11 above)
Comment (sachin-chamute) [reply ClariseMain/src/main/java/com/clairvoyant/clarise/controller/SkillsRatingController.java:31]: resolved

File: ClariseMain/src/main/java/com/clairvoyant/clarise/controller/SkillsRatingController.java (hunk 12 above)
Comment (sachin-chamute) [reply ClariseMain/src/main/java/com/clairvoyant/clarise/controller/SkillsRatingController.java]: no need of () and removed it
resolved

File: ClariseMain/src/main/java/com/clairvoyant/clarise/advice/ExceptionController.java (hunk 13 above)
Comment (sachin-chamute) [reply ClariseMain/src/main/java/com/clairvoyant/clarise/advice/ExceptionController.java]: added single ResourceNotFoundException for all resources instead of XxxNotFoundException

File: ClariseMain/src/main/java/com/clairvoyant/clarise/service/QualificationStatusService.java (hunk 4 above)
Comment (sachin-chamute) [reply ClariseMain/src/main/java/com/clairvoyant/clarise/service/QualificationStatusService.java:8]: followed DefaultQualificationStatusService this naming convention

**Review by santoshgaikwad14 — APPROVED** [review]
Looks ok to me
